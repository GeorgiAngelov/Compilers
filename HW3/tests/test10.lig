function f() {
      var x : [int] = [y, y, y];
      var y : int = sizeof(x);
}

function main(args : [[int]]) : int {
      var new_args : [[int]];
      var other_stuff : [bool];

      // Resize is polymorphic.
      new_args = resize(args, 10);
      // but doesn't accept nil.
      other_stuff = resize(nil, sizeof(args));
      
      new_args[0][0] = getchar();

      print(new_args[0]);

      return (new_args[0][0]);
}
