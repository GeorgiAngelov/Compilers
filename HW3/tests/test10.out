(decls
   (decl-fun:ok (id f) (type-fun (params) (type-void))
      (decls
         (decl-var:ok (id x) (type-array (type-int)) (array-lit:[int] (exps (id:int y) (id:int y) (id:int y))))
         (decl-var:ok (id y) (type-int) (fun-call:int (id sizeof) (exps (id:[int] x)))))
      (stmts))
   (decl-fun:NONE (id main) (type-fun (params (typed-id (id args) (type-array (type-array (type-int))))) (type-int))
      (decls
         (decl-var:ok (id new_args) (type-array (type-array (type-int))))
         (decl-var:ok (id other_stuff) (type-array (type-bool))))
      (stmts
         (assign:ok (id:[[int]] new_args) (fun-call:[[int]] (id resize) (exps (id:[[int]] args) (num:int 10))))
         (assign:NONE (id:[bool] other_stuff) (fun-call:NONE (id resize) (exps (nil:nil) (fun-call:int (id sizeof) (exps (id:[[int]] args))))))
         (assign:ok (array-idx:int (array-idx:[int] (id:[[int]] new_args) (num:int 0)) (num:int 0)) (fun-call:int (id getchar) (exps)))
         (exp-stmt:ok (fun-call:void (id print) (exps (array-idx:[int] (id:[[int]] new_args) (num:int 0)))))
         (return:ok (array-idx:int (array-idx:[int] (id:[[int]] new_args) (num:int 0)) (num:int 0))))))
Typechecking failed!
